src/Uno.UI/UI/Xaml/Controls/ComboBox/ComboBox.cs

```diff
@@ -50,10 +52,20 @@ protected override void OnApplyTemplate()
 		{
 			base.OnApplyTemplate();
 
+			if (_popup is PopupBase oldPopup)
+			{
+				oldPopup.CustomLayouter = null;
+			}
+
 			_popup = this.GetTemplateChild("Popup") as IPopup;
 			_popupBorder = this.GetTemplateChild("PopupBorder") as Border;
 			_contentPresenter = this.GetTemplateChild("ContentPresenter") as ContentPresenter;
 
+			if (_popup is PopupBase popup)
+			{
+				popup.CustomLayouter = new DropDownLayouter(this, popup);
```

Comment:
You need to create a new one each time?

Reply:
As it has a reference to `this` and the `popup` yes. We could find some way to avoid that, but the cost is really small, especially as we usually don't change the template of controls.